async-nats/tests/client_tests.rs

```diff
@@ -126,4 +126,62 @@ mod client {
         .unwrap();
         assert_eq!(resp.unwrap().payload, Bytes::from("reply"));
     }
+
+    #[tokio::test]
+    async fn token_auth() {
+        let server = nats_server::run_server("tests/configs/token.conf");
+        let mut nc = async_nats::ConnectOptions::with_token("s3cr3t".into())
+            .connect(server.client_url())
+            .await
+            .unwrap();
+
+        let mut sub = nc.subscribe("test".into()).await.unwrap();
+        nc.publish("test".into(), "test".into()).await.unwrap();
+        nc.flush().await.unwrap();
+        assert!(sub.next().await.is_some());
+    }
+
+    #[tokio::test]
+    async fn user_pass_auth() {
+        let server = nats_server::run_server("tests/configs/user_pass.conf");
+        let mut nc = async_nats::ConnectOptions::with_user_pass("derek".into(), "s3cr3t".into())
+            .connect(server.client_url())
+            .await
+            .unwrap();
+
+        let mut sub = nc.subscribe("test".into()).await.unwrap();
+        nc.publish("test".into(), "test".into()).await.unwrap();
+        nc.flush().await.unwrap();
+        assert!(sub.next().await.is_some());
+    }
+
+    #[tokio::test]
+    async fn user_pass_auth_wrong_pass() {
+        let server = nats_server::run_server("tests/configs/user_pass.conf");
+        let mut nc =
+            async_nats::ConnectOptions::with_user_pass("derek".into(), "bad_password".into())
+                .connect(server.client_url())
+                .await
+                .unwrap();
+
+        let mut errs = nc.errors_stream().await.unwrap();
```

Comment:
So this if this checking for auth errors  passed to connect, we definitively don't want to be doing that.
Connect should make sure that it was successful before returning imo.